scan/dependencies.go

```diff
@@ -127,7 +127,7 @@ func NormalizeImageTag(img string) string {
 
 // NewImageReference parses a path of a image and creates a ImageReference object
 func NewImageReference(imagePath string) (*image.Reference, error) {
-	ref, err := reference.Parse(imagePath)
+	ref, err := reference.Parse(util.TrimQuotes(imagePath))
```

Comment:
I added a comment to TrimQuotes in https://github.com/Azure/acr-builder/pull/546#discussion_r349727992

Reply:
we should handle the quote here instead:

https://github.com/Azure/acr-builder/blob/master/scan/dependencies.go#L202

Reply:
Also, there is a function in this file that does the same, you might wanna move that to Util and use that instead:

https://github.com/Azure/acr-builder/blob/master/scan/dependencies.go#L283

Your function below will fail in this case:

```go
	func TrimQuotes(s string) string {
	if strings.HasPrefix(s, "'") {
		return strings.Trim(s, "'")
	}
	return strings.Trim(s, "\"")
}
```

FROM alpine' 

Since it checks for Prefix. We can just use this instead:

```go
// removeSurroundingQuotes trims double quotes, then single quotes.
func removeSurroundingQuotes(s string) string {
	s = strings.Trim(s, `"`)
	return strings.Trim(s, `'`)
}
```

Reply:
@shahzzzam we can optimize by simply having
```
// removeSurroundingQuotes trims double quotes, then single quotes.
func removeSurroundingQuotes(s string) string {
	return strings.Trim(s, `'\"`)
}
```
See fix.

Reply:
@northtyphoon @shahzzzam the `removeSurroundingQuotes` method breaks in the case where the beginning and ending quotes are of different type. ie. `"'hello world"'` renders as `hello world"`. I will move this method to util instead with the fix

Reply:
@shahzzzam good suggestion. I will move the trim there when we first parse FROM